docs/HyperIndex/Guides/event-handlers.mdx

```diff
@@ -18,78 +18,43 @@ envio codegen
 
 in the project directory to generate the functions you will use in your handlers.
 
-Each event requires two functions to be registered:
+Each event requires that a handler be registerred. These functions are asyncronous.
 
-1. `loader` function
-1. `handler` function
+**Note:** _Two other functions are also sometimes used but not required. Both are explored in the Advanced section. One is used to optimise the loading of entities from the database in performance critical situations, the other is to be used for dynamic/factory contracts loading._
```

Comment:
> **Note:** _Two other functions are also sometimes used but not required.

It's not very clear what functions you're talking about here.

Reply:
```suggestion
**Note:** _Two other functions, `handlerWithLoader` and `contractRegister`, are also sometimes used but not required for most indexers. Both functions are explored in the Advanced section. `handlerWithLoader` is used to optimise the loading of entities from the database in performance critical situations, `contractRegister` is to be used for dynamic/factory contracts loading._
<!-- TODO: add links to the advanced section about handlerWithLoader and contractRegister -->
```